rosbag2_transport/CMakeLists.txt

```diff
@@ -99,144 +100,153 @@ ament_export_libraries(${PROJECT_NAME})
 ament_export_targets(export_${PROJECT_NAME})
 
 ament_export_dependencies(
-  rclcpp_components
   keyboard_handler
+  rclcpp
   rosbag2_cpp
-  rosbag2_compression
```

Comment:
Out of curiosity, why remove `rosbag2_compression` from dependencies? 
Implying that `rosbag2_compression` includes as dependencies for the `rosbag2_storage`?

Reply:
To be clear, we are only removing it from the *exported* dependencies.  That's because `rosbag2_compression` is only used in the source files, not the header files.

In other words, you only need to export dependencies that your header files depend on.  Downstream packages don't need to know about the internal dependencies.  In turn, this should make linking of downstream packages a bit faster.